packages/sanity/src/_internal/cli/actions/schema/fetchSchemaAction.ts

```diff
@@ -0,0 +1,41 @@
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import {type SanityDocumentLike} from '@sanity/types'
+
+export interface FetchSchemaFlags {
+  id: string
+}
+
+export default async function fetchSchemaAction(
+  args: CliCommandArguments<FetchSchemaFlags>,
+  context: CliCommandContext,
+): Promise<void> {
+  const {apiClient, output} = context
+  const flags = args.extOptions
+  const schemaId = flags.id
+  const spinner = output.spinner({}).start('Fetching schema')
+  const client = apiClient({
+    requireUser: true,
+    requireProject: true,
+  }).withConfig({apiVersion: 'v2024-08-01'})
+
+  const projectId = client.config().projectId
+  const dataset = client.config().dataset
+
+  const schemas = await client
+    .withConfig({
+      dataset: dataset,
+      projectId: projectId,
+    })
+    .fetch<SanityDocumentLike[]>(`*[_type == "sanity.workspace.schema" && _id == "${schemaId}"]`)
```

Comment:
The type should be a constant an reused troughout. You should also prefer using GROQ params in queries for variables, ie `fetch(*[_id==$schemaId], { schemaId})`